packages/service-library/tests/test_progress_bar.py

```diff
@@ -262,6 +262,33 @@ async def test_set_progress(caplog: pytest.LogCaptureFixture, faker: Faker):
         assert "TIP:" in caplog.messages[0]
 
 
+async def test_reset_progress(caplog: pytest.LogCaptureFixture, faker: Faker):
+    async with ProgressBarData(num_steps=50, description=faker.pystr()) as root:
+        assert root._current_steps == pytest.approx(0)  # noqa: SLF001
+        assert root.num_steps == 50
+        assert root.step_weights is None
+        await root.set_(50)
+        assert root._current_steps == pytest.approx(50)  # noqa: SLF001
+        assert "already reached maximum" not in caplog.text
+        await root.set_(51)
+        assert root._current_steps == pytest.approx(50)  # noqa: SLF001
+        assert "already reached maximum" in caplog.text
```

Comment:
I would actually check for any warning/error instead of a specific text that might change and this  test will fail